pycytominer_transform/convert.py

```diff
@@ -2,349 +2,254 @@
             )
+            .arrow()
         )
 
-    # if we haven't set a source datatype and need to rely on the inferred one
-    # set it so it may be returned
-    if source_datatype is None:
-        source_datatype = suffixes[0]
-
-    return source_datatype
-
-
-@task
-def filter_source_filepaths(
-    records: Dict[str, List[Dict[str, Any]]], source_datatype: str
-) -> Dict[str, List[Dict[str, Any]]]:
-    """
-    Filter source filepaths based on provided source_datatype
-
-    Args:
-      records: Dict[str, List[Dict[str, Any]]]
-        Grouped datasets of files which will be used by other functions.
-      source_datatype: str
-        Source datatype to use for filtering the dataset.
-
-    Returns:
-      Dict[str, List[Dict[str, Any]]]
-        Data structure which groups related files based on the datatype.
-    """
-
-    return {
-        filegroup: [
-            file
-            for file in files
-            # ensure the filesize is greater than 0
-            if file["source_path"].stat().st_size > 0
-            # ensure the datatype matches the source datatype
-            and file["source_path"].suffix == f".{source_datatype}"
-        ]
-        for filegroup, files in records.items()
-    }
-
-
-@flow
-def gather_records(
-    source_path: str,
-    source_datatype: Optional[str] = None,
-    compartments: Optional[List[str]] = None,
-    **kwargs,
-) -> Dict[str, List[Dict[str, Any]]]:
-
-    """
-    Flow for gathering records for conversion
-
-    Args:
-      source_path: str:
-        Where to gather file-based data from.
-      source_datatype: Optional[str]:  (Default value = None)
-        The source datatype (extension) to use for reading the tables.
-      compartments: Optional[List[str]]:  (Default value = None)
-        The source file names to target within the provided path.
-
-    Returns:
-      Dict[str, List[Dict[str, Any]]]
-        Data structure which groups related files based on the compartments.
-    """
-
-    source_path = build_path(path=source_path, **kwargs)
-
-    # gather filepaths which will be used as the basis for this work
-    records = get_source_filepaths(path=source_path, compartments=compartments)
-
-    # infer or validate the source datatype based on source filepaths
-    source_datatype = infer_source_datatype(
-        records=records, source_datatype=source_datatype
-    )
-
-    # filter source filepaths to inferred or source datatype
-    return filter_source_filepaths(records=records, source_datatype=source_datatype)
+    return source
 
 
 @task
-def read_file(record: Dict[str, Any]) -> Dict[str, Any]:
+def prepend_column_name(
+    source: Dict[str, Any],
+    source_group_name: str,
+    identifying_columns: Union[List[str], Tuple[str, ...]],
+    metadata: Union[List[str], Tuple[str, ...]],
+    targets: List[str],
+) -> Dict[str, Any]:
     """
-    Read csv file from record.
+    Rename columns using the source group name, avoiding identifying columns.
```

Comment:
I like your comments in the method body that show examples of how columns are renamed, but it'd be nice to see that reiterated here in the method documentation. At the moment it's unclear to me how someone would know from just reading the docs that `targets` needs to include a string that's contained in `source_group_name`, or even that `source_group_name` should be a filename.

Reply:
Thank you - great points! I've added in further notes within the docstring in [44f1402](https://github.com/cytomining/pycytominer-transform/pull/22/commits/44f1402e05526089ef5d9e478beea3a5d9eb31c7) and am marking this resolved.